app/models/incomeSourceDetails/IncomeSourceDetailsResponse.scala

```diff
@@ -99,6 +96,12 @@ case class IncomeSourceDetailsModel(nino: String,
       case _ => None
     }
   }
+
+  def compareHashToQueryString(incomeSourceIdHash: Option[IncomeSourceIdHash])(implicit user: MtdItUser[_]): Option[IncomeSourceId] = {
```

Comment:
I would suspect that this should be:

`def compareHashToQueryString(incomeSourceIdHash: IncomeSourceIdHash)(implicit user: MtdItUser[_]): Option[IncomeSourceId]`